acquire/outputs/tar.py

```diff
@@ -26,23 +42,61 @@ def __init__(self, path, compress=False, encrypt=False, public_key=None, **kwarg
         else:
             self.tar = tarfile.open(name=self.path, mode=mode)
 
-    def write(self, path, fh, size=None, entry=None):
+    def write(
+        self,
+        path: Path,
+        fh: BinaryIO,
+        size: Optional[int] = None,
+        entry: Optional[FilesystemEntry] = None,
+    ) -> None:
+        """Write a filesystem entry or file-like object to a tar file.
+
+        Args:
+            path: The path of the entry to write.
+            fh: The file-like object of the entry to write.
+            size: The optional file size in bytes of the entry to write.
+            entry: the optional filesystem entry of the entry to write.
+        """
+        stat = None
         size = size or getattr(fh, "size", None)
-        if size is None:
+
+        if size is None and fh.seekable():
             offset = fh.tell()
             fh.seek(0, io.SEEK_END)
             size = fh.tell()
             fh.seek(offset)
 
         info = self.tar.tarinfo()
-        info.name = path
+        info.name = str(path)
         info.uname = "root"
         info.gname = "root"
-        info.size = size
+        info.size = size or 0
+
+        if entry.is_symlink():
+            try:
+                info.type = tarfile.SYMTYPE
+                info.linkname = entry.readlink().as_posix()
+            except FileNotFoundError:
```

Comment:
I don't think this can error.

Reply:
This indeed does not seem error anymore.